File: 백준 문제 풀이 모음/class 3/11403.py
import sys

n = int(sys.stdin.readline())

graph = []
for i in range(n) :
    tmp = list(map(int, sys.stdin.readline().split()))
    graph.append(tmp)


for i in range(n) :
    for j in range(n) :
        if graph[i][j] == 0 :
            graph[i][j] = 1000000

for k in range(n) :
    for i in range(n) :
        for j in range(n) :
            if graph[i][j] > graph[i][k] + graph[k][j] :  
                graph[i][j] =  graph[i][k] + graph[k][j]  
                

for i in range(n) :
    for j in range(n) :
        if graph[i][j] == 1000000 :
            graph[i][j] = 0
        else :
            graph[i][j] = 1
        print(graph[i][j], end = ' ')
    print()

File: new_백준/11403.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter


fun main() {

    var br = BufferedReader(InputStreamReader(System.`in`))
    var bw = BufferedWriter(OutputStreamWriter(System.out))

    val n = br.readLine().toInt()

    val graph = mutableListOf<IntArray>()

    repeat(n) {
        val tempList = br.readLine().split(" ").map {
            if (it.toInt() == 0) {
                Integer.MAX_VALUE / 2
            } else {
                it.toInt()
            }
        }.toIntArray()
        graph.add(tempList)
    }


    for (k in 0 until n) {
        for (i in 0 until n) {
            for (j in 0 until n) {
                graph[i][j] = minOf(graph[i][j], graph[i][k] + graph[k][j])
            }
        }
    }

    for(i in 0 until n) {
        for(j in 0 until n) {
            print("${if(graph[i][j] == Integer.MAX_VALUE / 2 || graph[i][j] == 0) 0 else 1} ")
        }
        println()
    }




    bw.flush()
    bw.close()
}

